source/MaterialXGenShader/ShaderGraph.cpp

```diff
@@ -112,6 +112,15 @@ void ShaderGraph::addUpstreamDependencies(const Element& root, ConstMaterialPtr
         // Make connections
         //
 
+        // Find the output to connect to
+        OutputPtr nodeDefOutput = upstreamNode->getNodeDefOutput(edge);
+        ShaderOutput* output = nodeDefOutput ? newNode->getOutput(nodeDefOutput->getName()) : newNode->getOutput();
```

Comment:
Use the new method to get the output from the nodedef (if the node is a multioutput node, or has an explicitly named output). If no such output is used just pick the ordinary default output on the node instance.